File: 02-programacion-basica/14-modularizacion-depuracion-y-pruebas/scala/ejemplos-guiados/PruebasYDepuracionBasica.scala
object PruebasYDepuracionBasica:
  def maximo(a: Int, b: Int): Int =
    if a >= b then a else b

  def main(args: Array[String]): Unit =
    // assert detiene la ejecución si la condición comprobada es falsa.
    assert(maximo(9, 4) == 9, "valores distintos")
    assert(maximo(7, 7) == 7, "empate")
    assert(maximo(-2, -8) == -2, "negativos")

    println("Todos los casos pasaron correctamente.")

File: 02-programacion-basica/14-modularizacion-depuracion-y-pruebas/java/ejemplos-guiados/PruebasYDepuracionBasica.java
public class PruebasYDepuracionBasica {
    static int maximo(int a, int b) {
        return a >= b ? a : b;
    }

    static void comprobar(String caso, int esperado, int obtenido) {
        if (esperado != obtenido) {
            throw new AssertionError(caso + " | esperado=" + esperado + " obtenido=" + obtenido);
        }
        System.out.println("OK: " + caso);
    }

    public static void main(String[] args) {
        // Se prueban situaciones distintas; no solo el caso más común.
        comprobar("valores distintos", 9, maximo(9, 4));
        comprobar("empate", 7, maximo(7, 7));
        comprobar("negativos", -2, maximo(-2, -8));
    }
}
